assignment-client/src/entities/EntityTreeSendThread.cpp

```diff
@@ -233,11 +270,36 @@ void EntityTreeSendThread::startNewTraversal(const ViewFrustum& view, EntityTree
                     bool success = false;
                     AACube cube = entity->getQueryAACube(success);
                     if (success) {
-                        if (_traversal.getCurrentView().cubeIntersectsKeyhole(cube) &&
-                                (entity->getLastEdited() > timestamp ||
-                                    !_traversal.getCompletedView().cubeIntersectsKeyhole(cube))) {
-                            float priority = _conicalView.computePriority(cube);
-                            _sendQueue.push(PrioritizedEntity(entity, priority));
+                        if (_traversal.getCurrentView().cubeIntersectsKeyhole(cube)) {
+                            // See the DiffTraversal::First case for an explanation of the "entity is too small" check
+                            AABox entityBounds = entity->hasChildren() ? AABox(cube) : entity->getAABox(success);
+                            if (!success) {
+                                entityBounds = AABox(cube);
+                            }
+
+                            float renderAccuracy = calculateRenderAccuracy(_traversal.getCurrentView().getPosition(),
+                                                                           entityBounds,
+                                                                           octreeSizeScale,
+                                                                           lodLevelOffset);
+
+                            // Only send entities if they are large enough to see
+                            if (renderAccuracy > 0.0f) {
+                                if (entity->getLastEdited() > timestamp || !_traversal.getCompletedView().cubeIntersectsKeyhole(cube)) {
+                                    float priority = _conicalView.computePriority(entityBounds);
+                                    _sendQueue.push(PrioritizedEntity(entity, priority));
+                                } else {
+                                    // If this entity was skipped last time because it was too small, we still need to send it
+                                    float lastRenderAccuracy = calculateRenderAccuracy(_traversal.getCompletedView().getPosition(),
+                                                                                       entityBounds,
+                                                                                       octreeSizeScale,
+                                                                                       lodLevelOffset);
+
+                                    if (lastRenderAccuracy <= 0.0f) {
+                                        float priority = _conicalView.computePriority(entityBounds);
+                                        _sendQueue.push(PrioritizedEntity(entity, priority));
+                                    }
+                                }
+                            }
```

Comment:
This logic looks correct, but after reading it... it occurred to me that there is a race condition that we'll eventually have to deal with:

The entity was visible and large enough to be sent in the `completedView` but has changed recently: it has become too small to see in the `currentView`  -->  we won't send an update --> it will remain large on the interface-client.  This is very similar to the "subsequent traversal failure" case I mentioned in the pensentation slides.  Not a blocker right now since the problem currently exists in `hifi/master`, but something to think about.